Tasks/AzureFunctionOnKubernetesV0/task.json

```diff
@@ -0,0 +1,123 @@
+            "label": "Docker registry service connection",
+            "groupName": "serviceConnections",
+            "helpMarkDown": "Select a Docker registry service connection."
+        },
+        {
+            "name": "kubernetesServiceConnection",
+            "type": "connectedService:kubernetes",
+            "label": "Kubernetes service connection",
+            "required": true,
+            "groupName": "serviceConnections",
+            "helpMarkDown": "Select a Kubernetes service connection."
+        },
+        {
+            "name": "secretName",
+            "type": "string",
+            "label": "Secret Name",
+            "groupName": "commands",
+            "helpMarkDown": "Kubernetes secret containing function config data (for ex. AzureWebJobsStorage: <Azure storage connection string>)"
+        },
+        {
+            "name": "namespace",
+            "type": "string",
+            "label": "Kubernetes namespace",
+            "groupName": "commands",
+            "helpMarkDown": "Kubernetes namespace"
+        },
+        {
+            "name": "appName",
+            "type": "string",
+            "required": true,
+            "label": "Application Name",
+            "groupName": "commands",
+            "helpMarkDown": "Application Name. The Kubernetes objects created use this name. This should follow Kubernetes naming conventions for resource names."
+        },
+        {
+            "name": "imageName",
+            "type": "string",
+            "label": "Image Name",
+            "groupName": "commands",
+            "helpMarkDown": "Image Name. If provided, the image is used for deployment. Otherwise, build and push (from the Dockerfile in the function root directory) and use that image."
+        },
+        {
+            "name": "registry",
+            "type": "string",
+            "label": "Registry",
+            "groupName": "commands",
+            "helpMarkDown": "Docker Hub namespace for private Docker repository / Registry url for other registries. Ex:<br><br>Docker Hub: billuser<br>ACR: contoso.azurecr.io<br><br>(Registry input is mandatory for private Docker Hub repository. For other registries, it's not required if the Docker service connection is provided)."
+        },
+        {
+            "name": "functionRootDirectory",
+            "type": "filePath",
+            "label": "Function root directory",
+            "required": false,
+            "groupName": "commands",
+            "helpMarkDown": "Function root directory. Should contain host.json. When image name is not provided, Docker build and push is performed from this directory."
+        },
+        {
+            "name": "waitForStability",
+            "type": "boolean",
+            "label": "Wait for stability",
+            "groupName": "commands",
+            "defaultValue": "false",
+            "helpMarkDown": "Wait for the Kubernetes objects to reach the desired state."
+        },
```

Comment:
Does this have to be an optional parameter? If so I'd suggest renaming it to something like `trigger: true` or something like that

Reply:
Will check with Shashank once.

Reply:
Discussed with Shashank. He is in favor of keeping this optional, default to true. And he was fine with the name.